File: app/src/main/java/com/alfianyusufabdullah/chatyuk/utils/InputTextListener.java
package com.alfianyusufabdullah.chatyuk.utils;

import android.support.design.widget.TextInputLayout;
import android.text.Editable;
import android.text.TextWatcher;

/**
 * Created by JonesRandom on 9/15/17.
 *
 * @site www.androidexample.web.id
 * @github @alfianyusufabdullah
 *
 */

public class InputTextListener implements TextWatcher {

    private TextInputLayout input;

    public InputTextListener(TextInputLayout input) {
        this.input = input;
    }

    @Override
    public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {

    }

    @Override
    public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
        input.setError("");
        input.setErrorEnabled(false);
    }

    @Override
    public void afterTextChanged(Editable editable) {

    }
}

File: app/src/main/java/com/alfianyusufabdullah/chatyuk/utils/InputTextListener.kt
package com.alfianyusufabdullah.chatyuk.utils

import com.google.android.material.textfield.TextInputLayout
import android.text.Editable
import android.text.TextWatcher

/**
 * Created by JonesRandom on 9/15/17.
 *
 * @site www.androidexample.web.id
 * @github @alfianyusufabdullah
 */

class InputTextListener(private val input: TextInputLayout) : TextWatcher {

    override fun beforeTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {

    }

    override fun onTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {
        input.error = ""
        input.isErrorEnabled = false
    }

    override fun afterTextChanged(editable: Editable) {

    }
}
